### PR TITLE
refactor(profile): convert from class to functional component - FE-3661

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
This PR converts Profile component from a class based to functional

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Do a regression for Profile component

### CODE REVIEW

File: src/components/profile/profile.component.js [hunk 1]
```diff
@@ -17,79 +17,73 @@ const marginPropTypes = filterStyledSystemMarginProps(
...
-      >
-        {this.avatar}
-        {this.text}
-      </ProfileStyle>
-    );
-  }
-}
+  return (
+    <ProfileStyle
+      large={large}
```
Comment (edleeks87) [inline src/components/profile/profile.component.js:76]: What does this prop do? It isn't used in the styled-component as far as I can tell?

File: src/components/profile/profile.spec.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import React from "react";
-import TestUtils from "react-dom/test-utils";
 import { shallow, mount } from "enzyme";
 import TestRenderer from "react-test-renderer";
```
Comment (edleeks87) [inline src/components/profile/profile.spec.js:3]: I think this can be removed as the test that uses it seems redundant as setting the `large` prop doesn't seem to do anything as far as I can tell

File: src/components/profile/profile.component.js [hunk 3]
```diff
@@ -17,79 +17,63 @@ const marginPropTypes = filterStyledSystemMarginProps(
...
-        {this.avatar}
-        {this.text}
-      </ProfileStyle>
-    );
-  }
-}
+  return (
+    <ProfileStyle
+      className={classes()}
+      hasSrc={hasSrc()}
```
Comment (grabkowski) [inline src/components/profile/profile.component.js:68]: do we need a method here? Couldnt it just be:
`hasSrc={!!src}`

File: src/components/profile/profile.component.js [hunk 4]
```diff
@@ -17,79 +17,63 @@ const marginPropTypes = filterStyledSystemMarginProps(
...
-      >
-        {this.avatar}
-        {this.text}
-      </ProfileStyle>
-    );
-  }
-}
+  return (
+    <ProfileStyle
+      className={classes()}
```
Comment (grabkowski) [inline src/components/profile/profile.component.js:67]: why not just `className={classNames(className)}`?

File: src/components/profile/profile.component.js [hunk 5]
```diff
@@ -103,14 +87,8 @@ Profile.propTypes = {
   email: PropTypes.string.isRequired,
   /** Define initials to display if there is no Gravatar image. */
   initials: PropTypes.string,
-  /** [Legacy] Enable a larger theme for the name. */
-  large: PropTypes.bool,
```
Comment (grabkowski) [inline src/components/profile/profile.component.js:107]: that should be done as a separate commit with it's own breaking change

**Review by grabkowski — CHANGES_REQUESTED** [review]

File: src/components/profile/profile.component.js (hunk 5 above)
Comment (mkrds) [reply src/components/profile/profile.component.js:107]: It's not really a breaking change as `large` doesn't do anything

**Review by grabkowski — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]
